ext/oj/strict.c

```diff
@@ -35,18 +35,22 @@ VALUE oj_calc_hash_key(ParseInfo pi, Val parent) {
         if (Yes == pi->options.sym_key) {
             rkey = ID2SYM(rb_intern3(parent->key, parent->klen, oj_utf8_encoding));
         } else {
+#if HAVE_RB_ENC_INTERNED_STR
+            rkey = rb_enc_interned_str(parent->key, parent->klen, oj_utf8_encoding);
+#else
             rkey = rb_str_new(parent->key, parent->klen);
             rkey = oj_encode(rkey);
+            OBJ_FREEZE(rkey);
+#endif
         }
-        OBJ_FREEZE(rkey);
         return rkey;
     }
     if (Yes == pi->options.sym_key) {
 	rkey = oj_sym_intern(parent->key, parent->klen);
     } else {
 	rkey = oj_str_intern(parent->key, parent->klen);
-    }
```

Comment:
Line 48 should not be removed and line 543 should not be added. Both `oj_str_intern` already  freezes the strings. As for `oj_sym_intern`, it does not only because my understanding is that symbols are immutable. If that is not the case then a freeze is needed in the oj intern.c and not here.

Reply:
My error. Misread the change. Good catch.